### PR TITLE
Replace underscore/lodash references in _.any() and _.isequal() with natives

### LINKED ISSUE #158: Remove underscore
Good first issue for [Hacktoberfest](https://hacktoberfest.com/). Remove the need for `underscore` package.

Good reference:
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore

### PR DESCRIPTION
**Pull request by Nandika-A**
Resolves #158 

Underscore and lodash references in lib/subscription.js in _.any() and _.isEqual() are replaced.
1. _.any(): replaced with [Array.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)
2. _.isEqual(): This method is imported as a Node module using `npm i --save lodash.isequal` as mentioned in the documentation. [link](https://www.npmjs.com/package/lodash.isequal?activeTab=code).  This module is imported in subscription.js as a package `var isEqual = require('lodash.isequal');`.

### CODE REVIEW

File: lib/subscription.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import { _ } from 'meteor/underscore'
 import DocumentRefCounter from './doc_ref_counter'
 import { debugLog } from './logging'
 
+var isEqual = require('lodash.isequal');
```
Comment (StorytellerCZ) [inline lib/subscription.js:6]: We can do the normal `import` syntax here.

File: package.json [hunk 2]
```diff
@@ -32,5 +32,8 @@
     "babel-eslint": "^10.1.0",
     "husky": "^8.0.3",
     "standard": "^17.1.0"
+  },
+  "dependencies": {
+    "lodash.isequal": "^4.5.0"
```
Comment (StorytellerCZ) [inline package.json:37]: We will need the `Npm.require` in `package.js` or https://atmospherejs.com/tmeasday/check-npm-versions
here to make this work properly. Probably the first one.

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Nandika-A) [reply package.json:37]: I have now replaced require() in subscription.js with normal import syntax.

**Comment by StorytellerCZ** [comment]
Fixed the imports. @jankapunkt ready for review.

**Review by StorytellerCZ — APPROVED** [review]

File: package.js [hunk 3]
```diff
@@ -6,6 +6,10 @@ Package.describe({
   git: 'https://github.com/Meteor-Community-Packages/meteor-publish-composite'
 })
 
+Npm.depends({
+    "lodash.isequal": "^4.5.0"
```
Comment (jankapunkt) [inline package.js:10]: Indent 2 chars, please

**Review by jankapunkt — CHANGES_REQUESTED** [review]

**Comment by StorytellerCZ** [comment]
The tests are failing, but from my testing this is not related to this PR, but to the something with async and that the tests need update.

File: lib/utils.js
Comment (jankapunkt) [inline lib/utils.js:1]: This file is empty by intention?

Comment (jankapunkt) [reply lib/utils.js:1]: Apart from this one the PR is ready to me

**Pull request merged**